src/test/java/seedu/address/model/band/UniqueBandListTest.java

```diff
@@ -0,0 +1,182 @@
+package seedu.address.model.band;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalBands.ACE;
+import static seedu.address.testutil.TypicalBands.BOOM;
+import static seedu.address.testutil.TypicalBands.CANDY;
+import static seedu.address.testutil.TypicalBands.ELISE;
+import static seedu.address.testutil.TypicalMusicians.ALICE;
+import static seedu.address.testutil.TypicalMusicians.ELLE;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.band.exceptions.BandNotFoundException;
+import seedu.address.model.band.exceptions.DuplicateBandException;
+import seedu.address.model.musician.exceptions.DuplicateMusicianException;
+import seedu.address.testutil.BandBuilder;
+
+public class UniqueBandListTest {
+
+    private final UniqueBandList uniqueBandList = new UniqueBandList();
+
+    @Test
+    public void contains_nullBand_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueBandList.contains(null));
+    }
+
+    @Test
+    public void contains_bandNotInList_returnsFalse() {
+        assertFalse(uniqueBandList.contains(ACE));
+    }
+
+    @Test
+    public void contains_bandInList_returnsTrue() {
+        uniqueBandList.add(ACE);
+        assertTrue(uniqueBandList.contains(ACE));
+    }
+
+    @Test
+    public void add_nullBand_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueBandList.add(null));
+    }
+
+    @Test
+    public void add_duplicateBand_throwsDuplicateBandException() {
+        uniqueBandList.add(ACE);
+        assertThrows(DuplicateBandException.class, () -> uniqueBandList.add(ACE));
+    }
+
+    @Test
+    public void add_duplicateBandWithNewMusician_throwsDuplicateBandException() {
+        uniqueBandList.add(ACE);
+        Band newAce = new BandBuilder(ACE).withMusicians(ALICE)
+            .build();
+        assertThrows(DuplicateBandException.class, () -> uniqueBandList.add(newAce));
```

Comment:
A good test case not included in the UniqueMusicianList but relevant to the UniqueBandList!